src/detect-file-data.c

```diff
@@ -377,12 +377,19 @@ static InspectionBuffer *HttpServerBodyGetDataCallback(DetectEngineThreadCtx *de
         if (swf_file_type == FILE_SWF_ZLIB_COMPRESSION ||
             swf_file_type == FILE_SWF_LZMA_COMPRESSION)
         {
-            (void)FileSwfDecompression(data, data_len,
-                                       det_ctx,
-                                       buffer,
-                                       htp_state->cfg->swf_compression_type,
-                                       htp_state->cfg->swf_decompress_depth,
-                                       htp_state->cfg->swf_compress_depth);
+            int ret = FileSwfDecompression(data, data_len, det_ctx, buffer,
+                    htp_state->cfg->swf_compression_type, htp_state->cfg->swf_decompress_depth,
+                    htp_state->cfg->swf_compress_depth);
+            if (ret == 0) {
+                AppLayerDecoderEvents *decoder_events = DetectEngineGetEvents(det_ctx);
+                if (det_ctx->p && decoder_events) {
+                    for (int i = 0; i < decoder_events->cnt; i++) {
```

Comment:
w/o looking deeper I wonder if we can just pass an `AppLayerDecoderEvents` pointer to `FileSwfDecompression`? Then we could pass the `p->app_layer_events` directly instead of using this copy logic.

Reply:
`FileSwfDecompression` can set events;  events can also be set by the functions it invokes -- notably `FileSwfZlibDecompression` and `FileSwfLzmaDecompression`.

So, I'm in favor of leaving the "copy logic" in place instead.

Another possible approach is to change `DetectEngineSetEvent` to apply the event to `p_app_layer_events` directly.